GoRogue/MapGeneration/Region.cs

```diff
@@ -71,13 +71,22 @@ public override string ToString()
         /// </summary>
         /// <param name="obj"/>
         /// <returns/>
-        public override bool Equals(object obj)
+        public override bool Equals(object obj) => obj is Region region && this == region;
+
+        /// <summary>
+        /// To facilitate in equality operations
+        /// </summary>
+        /// <returns>The Hash Code of the Region</returns>
+        /// <remarks>
+        /// Due to the nature of Connections being personal to the map generation algorithm,
+        /// connections are not considered when evaluating a region's Hash Code.
+        /// </remarks>
+        public override int GetHashCode()
         {
-            if (obj.GetType() != typeof(Region))
-                return false;
-            else
-                return this == (Region)obj;
+            return SouthEastCorner.GetHashCode() + NorthEastCorner.GetHashCode() + SouthWestCorner.GetHashCode() +
+                       NorthWestCorner.GetHashCode() + Center.GetHashCode();
```

Comment:
The values used to generate the hash code aren't readonly, which will generate a warning in most analyzers.  Unfortunately, there is no way to create a hash using only readonly fields, so this is fine; however when this is the case best practice is to clearly document either in the class documentation, the GetHashCode documentation, or both that if instances of this class are used as keys in a Dictionary or used in a HashSet, they _must not_ be modified while they are added to that data structure.  Having an object change hash code after it is added to such an object is undefined behavior.

Reply:
I fixed this by making those properties readonly. The tradeoff is that `Region`s can no longer perform transformations on themselves; transformations must always return a new region.